jax/_src/numpy/lax_numpy.py

```diff
@@ -3897,13 +3915,11 @@ def _eliminate_deprecated_list_indexing(idx):
   return idx
 
 def _is_boolean_index(i):
-  try:
-    abstract_i = core.get_aval(i)
-  except TypeError:
-    abstract_i = None
-  return (isinstance(abstract_i, ShapedArray) and issubdtype(abstract_i.dtype, bool_)
-          or isinstance(i, list) and i and _all(_is_scalar(e)
-          and issubdtype(_dtype(e), np.bool_) for e in i))
+  if isinstance(i, core.Tracer):
+    i = i.aval
```

Comment:
Why did you switch to an instance check here but continue using `try: core.get_aval()` in other places?

Reply:
`core.get_aval()` will always create a `DeviceArray` if the argument is not a `Tracer`. This is unnecessary when calling it with a scalar Python index and is also quite a bit slower than using an `isinstance` check:
```python
In [3]: %timeit core.get_aval(3)
8.94 µs ± 260 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [4]: %timeit isinstance(3, core.Tracer)
67.2 ns ± 0.969 ns per loop (mean ± std. dev. of 7 runs, 10,000,000 loops each)
```

Since these checks happen quite often this was noticeable in the performance profile, so I tried to remove it where needed. I left calls to `core.get_aval` untouched in all other code paths to keep this PR limited to indexing with a single Python indices, but I'm happy take a look at optimising other forms of constant indexing in a future PR if that would be appreciated.